sqlglot/generator.py

```
@@ -1670,8 +1670,14 @@ def properties_sql(self, expression: exp.Properties) -> str:
             elif p_loc == exp.Properties.Location.POST_SCHEMA:
                 root_properties.append(p)
 
-        root_props = self.root_properties(exp.Properties(expressions=root_properties))
-        with_props = self.with_properties(exp.Properties(expressions=with_properties))
+        root_props_ast = exp.Properties(expressions=root_properties)
+        root_props_ast.parent = expression.parent
```

Is there any scenario / odd case in which we forget to link the parent at some place and `UNIQUE` gets silently dropped? Is this the only place we need it or is our test coverage guaranteeing this would be caught?

---

No, the reason my approach didn't work was because the parent link got silently dropped due to `Properties(expressions=...)`. This is a one-off thing that is only relevant to the generator and is meant to enable stuff like what I did for the unique property.